gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManager.java

```diff
@@ -764,6 +777,12 @@ private void cleanUp() {
       }
 
       for (String dagId: dagIdstoClean) {
+        // send an event before cleaning up dag metadata
+        if (this.eventSubmitter.isPresent()) {
+          JobExecutionPlan jobExecutionPlan = this.dags.get(dagId).getNodes().get(0).getValue();
+          Map<String, String> jobMetadata = TimingEventUtils.getJobMetadata(Maps.newHashMap(), jobExecutionPlan);
+          this.eventSubmitter.get().getTimingEvent(TimingEvent.FlowTimings.FLOW_CANCEL).stop(jobMetadata);
```

Comment:
Not every dag in dagIdsToClean is in CANCELLED state. Should lines 780-784 be moved to a separate method (e.g. emitFlowEvent()) in DagManagerUtils and be invoked after line 772? And since we are treating cancelled flow as failed for now, emit FLOW_FAILED event?